internal/reconciliations/gateway/gateway.go

```diff
@@ -54,39 +56,46 @@ func ReconcileKymaGateway(ctx context.Context, k8sClient client.Client, apiGatew
 	ctrl.Log.Info("Reconcile Kyma Gateway", "enabled", apiGatewayCR.Spec.EnableKymaGateway)
 	if isKymaGatewayEnabled(*apiGatewayCR) && !apiGatewayCR.IsInDeletion() && !hasKymaGatewayFinalizer(*apiGatewayCR) {
 		if err := addKymaGatewayFinalizer(ctx, k8sClient, apiGatewayCR); err != nil {
-			return controllers.ErrorStatus(err, "Failed to add finalizer during Kyma Gateway reconciliation")
+			return controllers.ErrorStatus(err, "Failed to add finalizer during Kyma Gateway reconciliation", conditions.KymaGatewayReconcileFailed.Condition())
 		}
 	}
 
 	if !hasKymaGatewayFinalizer(*apiGatewayCR) {
 		ctrl.Log.Info("There is no Kyma Gateway finalizer, skipping reconciliation")
-		return controllers.ReadyStatus()
+		return controllers.ReadyStatus(conditions.KymaGatewayReconcileSucceeded.Condition())
 	}
 
 	if !isKymaGatewayEnabled(*apiGatewayCR) || apiGatewayCR.IsInDeletion() {
 		resourceFinder, err := resources.NewResourcesFinderFromConfigYaml(ctx, k8sClient, ctrl.Log, apiGatewayResourceListPath)
 		if err != nil {
-			return controllers.ErrorStatus(err, "Could not read customer resources finder configuration")
+			return controllers.ErrorStatus(err, "Could not read customer resources finder configuration", conditions.KymaGatewayReconcileFailed.Condition())
 		}
 
 		clientResources, err := resourceFinder.FindUserCreatedResources(checkDefaultGatewayReference)
 		if err != nil {
-			return controllers.ErrorStatus(err, "Could not get customer resources from the cluster")
+			return controllers.ErrorStatus(err, "Could not get customer resources from the cluster", conditions.KymaGatewayReconcileFailed.Condition())
 		}
 
 		if len(clientResources) > 0 {
-			for _, res := range clientResources {
+			var blockingResources []string
+			for i := 0; i < 5; i++ {
+				if i >= len(clientResources) {
+					break
+				}
+				res := clientResources[i]
 				ctrl.Log.Info("Custom resource is blocking Kyma Gateway deletion", "gvk", res.GVK.String(), "namespace", res.Namespace, "name", res.Name)
```

Comment:
Do we need to move this log outside of the loop to be reliable? With the current implementation we will only log the first 5 blocking resources, but we want to have all in the log.

Reply:
Do you? The implementation was only about first 5. If there are going to be 1000 blocking elements, your logs will be pretty much cluttered. I can modify it, so it goes over all elements but only collects 5 elements but I would avoid cluttering the logs.